src/Client.php

```diff
@@ -42,15 +42,15 @@ public function __construct(ClientOptions $options)
         $this->eventHandler = new EventHandler();
     }
 
-    public function connect(): void
+    public function connect(?callable $execute = null): void
```

Comment:
Thanks for your pull request! 🎉

I think adding this to the connect method will require a major release, due backwards incompatible changes, which is planned in a few months.

I prefer Closure > callable. And I would like to keep all callable a Closure in this project.

If you want use callable you can use Closure::from callable(...) within your project.

https://www.php.net/manual/en/closure.fromcallable.php

Futuremore, you may want to add this feature to the official documentation (if it's get merged):

https://github.com/ghostzero/tmi-website/edit/main/docs/docs/README.md

Everything else looks good 😀

Reply:
> I think adding this to the connect method will require a major release, due backwards incompatible changes, which is planned in a few months.

Would you prefer a `execute()` method instead to leave the connect method as-is? You will call `execute(closure)` that, behind the scenes, calls `connect()`.

> I prefer Closure > callable. And I would like to keep all callable a Closure in this project.
No problem! I'll change it :)

Reply:
Sorry for the long delay, i've hd to done some other things first. I also was thinking of the `callable` think. Maybe it would be better to use callable :thinking:

Reply:
Np, life has always the priority :) Glad to see you merged this, thank you!